milestone2

```diff
@@ -0,0 +1,82 @@
+import java.util.HashMap;
+import java.util.Iterator;
+import java.util.LinkedList;
+import java.util.Scanner;
+
+public class ToDoList<T> {
+    private LinkedList<T> tasks; // Use LinkedList
+    private HashMap<T, String> categories; // Associative data structure
+
+    public ToDoList() {
+        tasks = new LinkedList<>();
+        categories = new HashMap<>();
+    }
+
+    public void addTask(T task, String category) {
+        tasks.add(task);
+        categories.put(task, category);
```

Comment:
Might be more useful to implement the other way around: Map categories to a list of tasks, so that when you want to look up a category you can retrieve all tasks that are in that category.